### PR TITLE
Implemented Email queries and some small adjustments

### PR DESCRIPTION
**Pull request by iaronaraujo**
Implemented send mail to all teachers from an institution, all students, all students in a grade and all students in a subject

Small adjustments:
- User now have "lastInstitutionId"
- Added checks to advanced entry code values
- Added quantity of students and teachers in a grade
- Fixed content size of reply and discussion

### CODE REVIEW

File: src/main/java/br/edu/ufcg/genus/graphql/queries/EmailQueries.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		return emailService.sendEmailToTeachers(institutionId, user, input.getSubject(), input.getText());	
+		
+	}
+	
+	public Boolean sendEmailToStudents(SendEmailInput input, Long institutionId) {
+		User user = userService.findLoggedUser();
+		List<UserRole> permitedRoles = new ArrayList<>();
+		permitedRoles.add(UserRole.ADMIN);
+		PermissionChecker.checkPermission(user, institutionId, permitedRoles);
+		return emailService.sendEmailToTeachers(institutionId, user, input.getSubject(), input.getText());
```
Comment (mari-linhares) [inline src/main/java/br/edu/ufcg/genus/graphql/queries/EmailQueries.java]: emailService.sendEmailToTeachers -> emailService.sendEmailToStudents

File: src/main/java/br/edu/ufcg/genus/inputs/EvaluationEditInput.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
+package br.edu.ufcg.genus.inputs;
+
+public class EvaluationEditInput {
+	
+	private long evaluationId;
+	private String name;
+	private Double result;
+	private Double weight;
+	
+	public EvaluationEditInput() {
```
Comment (mari-linhares) [inline src/main/java/br/edu/ufcg/genus/inputs/EvaluationEditInput.java]: Is this empty constructor really needed?

File: src/main/java/br/edu/ufcg/genus/inputs/SendEmailInput.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package br.edu.ufcg.genus.inputs;
+
+public class SendEmailInput {
+	
+	private String subject;
+	
+	private String text;
+	
+	public SendEmailInput() {
```
Comment (mari-linhares) [inline src/main/java/br/edu/ufcg/genus/inputs/SendEmailInput.java]: Same, is this needed?

File: src/main/java/br/edu/ufcg/genus/services/SubjectService.java [hunk 4]
```diff
@@ -256,5 +265,18 @@ public boolean removeTeacherFromSubject(Long subjectId, Long teacherId, User use
...
 		return result;
 	}
+	
+	public Subject moveStudentsFromSubject(Long fromId, Long toId, User user) {
+		List<Long> studentIds = new ArrayList<>();
+		Subject sub = findSubjectById(fromId);
+		for(StudentSubject studSub: sub.getStudents()) {
+			studentIds.add(studSub.getUser().getId());
+		}
+		removeEveryStudentFromSubject(fromId, user);
```
Comment (mari-linhares) [inline src/main/java/br/edu/ufcg/genus/services/SubjectService.java]: I think we should not remove from the original subject, right?

**Comment by iaronaraujo** [comment]
Problems fixed

**Pull request merged**